### PR TITLE
Cache queries

### PR DESCRIPTION
**Pull request by Iiridayn**
Caches the item lookup queries to match the API cache time - saves the RTT and server load.

### CODE REVIEW

**Comment by Iiridayn** [comment]
fixes: #17

File: src/main.rs [hunk 1]
```diff
@@ -67,21 +69,33 @@ struct Opt {
...
+    )
+});
+
+static DATA_DIR_HELP: Lazy<String> = Lazy::new(|| {
     format!(
         r#"Save cached recipes and items to this directory
 
 If provided, the parent directory of the cache directory must already exist. Defaults to '{}'."#,
-        cache_dir().unwrap().display()
+        data_dir(&None).unwrap().display()
```
Comment (t-mw) [inline src/main.rs:98]: I wonder if an additional directory is necessary? We could still call the reset option `reset_data` even with a single `cache_dir` option, and maybe doc it as something like `Cache API responses, including recipe and item data, in this directory `

**Review by t-mw — COMMENTED** [review]
Great, thank you! Only wondering whether we could simplify and still use one directory.

File: src/main.rs (hunk 1 above)
Comment (Iiridayn) [reply src/main.rs:98]: Yes, an additional directory _is_ necessary. For users like myself who backup files, it is very important to put the kind of data into the location it matches. Automatic backup scripts can then store important files (config), and useful files with a lower priority (such as these data files), while ignoring cache files.

The `dirs` module makes it trivial to do the right thing here, across every system. I cannot in good conscience not fix this while touching the code.

Comment (Iiridayn) [reply src/main.rs:98]: I would also note that on Windows, these two directories are the same, if we use `data_local_dir`. Really, the question is not "how can we keep all the data in the same place?" and more "how valuable do we think this data is?" - then use the corresponding directory for the data.

File: src/request.rs [hunk 2]
```diff
@@ -149,31 +155,86 @@ where
...
+    T: serde::de::DeserializeOwned,
+{
+    // Hash URL, check if cached, deserialize if so
+    // We're hashing on url since the API does as well
+    let mut hash = DefaultHasher::new();
+    url.hash(&mut hash);
+    let hash = hash.finish();
+
+    let mut cache_file = cache_dir.clone();
+    cache_file.push(format!("cache_{}", hash));
```
[the remark this answers is not in the document]
Comment (Iiridayn) [reply src/request.rs]: Good idea - done.

File: src/main.rs (hunk 1 above)
Comment (t-mw) [reply src/main.rs:98]: For me the files being stored in the data directory are also cache files, but it's fine

**Review by t-mw — APPROVED** [review]

**Comment by Iiridayn** [comment]
This was merged (w/the single truth cache prefix) as part of #24, so closing this PR.

**Pull request closed without merging**